### PR TITLE
v1.0.0-rc1

### LINKED ISSUE #1430: Grunt v1.0.0-rc1
This release is to address critical issues, migrate to the jQuery Foundation and communication about the future of Grunt.

Specifically the issues we are addressing in this release are: https://github.com/gruntjs/grunt/milestones/rc1

- [ ] ~~Move to jQuery Foundation~~ 👉`1.0.0`
- [x] Create a rfcs repo: https://github.com/gruntjs/rfcs
- [x] Ensure compatibility with Node.js v0.10, v0.12, 4.x and latest npm
- [x] Drop support for Node.js v0.8
- [x] Update outdated dependencies and provide an upgrade path for breaking changes
  - [x] Update `grunt-legacy-util` dependencies
    - [x] `grunt.util._.template` now returns a compile function so `template.process` needs to be updated to accommodate `tmpl = grunt.util._.template(tmpl, options)(data);`.  
    Fixed in `next/grunt-legacy-util` 3cc26ba4caf9d673fddf4f4c4c4d3f2446a7d319
    - [x] `grunt.util._.flatten` doesn't work the same. It's now `flattenDeep` and needs to be updated in certain places to ensure compatibility.  
    Fixed in `next/grunt-legacy-util` 3cc26ba4caf9d673fddf4f4c4c4d3f2446a7d319
  - [x] `coffee-script@1.10.x`
  - [x] `iconv-lite@0.4.x` (Strips BOM by default, updated to provide `stripBOM: !file.preserveBOM`)
  - [x] `glob@6.0.x`
  - [x] `dateformat@1.0.12` (~~look into why this is tagged differently~~ Because of old fashion versioning)
  - [x] `minimatch@3.0.x`
  - [x] `js-yaml@3.5.x`
  - [x] `nopt@3.0.x`
- [x] Add bin script to local Grunt
  - [ ] ~~Add note about: `npm run grunt -- task:target` is possible `npm >= 2.0.0`~~ 👉`1.0.0`
- [x] Update `grunt-cli` dependencies
- [x] Investigate why glob/minimatch doesn't seem to sort on Node.js v0.10 (https://github.com/isaacs/node-glob/issues/215)
- [x] Extend `grunt.util._.templateSettings` instead of overriding: GH-1151
- [x] `mode` can be passed to `grunt.file.write`. GH-956

### PR DESCRIPTION
**Pull request by vladikoff**
Fixes #1430

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -40,9 +40,9 @@
     "exit": "~0.1.1",
     "findup-sync": "~0.3.0",
     "glob": "~6.0.4",
-    "grunt-cli": "git://github.com/gruntjs/grunt-cli",
-    "grunt-legacy-log": "~0.1.0",
-    "grunt-legacy-util": "git://github.com/gruntjs/grunt-legacy-util",
+    "grunt-cli": "1.0.0-rc1",
+    "grunt-legacy-log": "1.0.0-rc1",
+    "grunt-legacy-util": "1.0.0-rc1",
```
Comment (vladikoff) [inline package.json]: @shama these are TODO before release. I think that's all left todo

**Comment by vladikoff** [comment]
Will need to restart the build after https://github.com/gruntjs/grunt/pull/1457#discussion_r52550390 is done.

File: CHANGELOG [hunk 2]
```diff
@@ -1,3 +1,29 @@
...
+    - `coffee-script` is upgraded to `~1.10.0` which could incur breaking changes
+      when using the language with plugins and Gruntfiles.
+    - `nopt` is upgraded to `~3.0.6` which has fixed many issues, including passing
+      multiple arguments and dealing with numbers as options. Be aware previously
+      `--foo bar` used to pass the value `'bar'` to the option `foo`. It will now
+      set the option `foo` to `true` and run the task `bar`.
+    -`glob` is upgraded to `~6.0.4` and `minimatch` is upgraded to `~3.0.0`. Results
+      are now sorted by default with `grunt.file.expandMapping()`. Pass the
+      `nosort: true` option if you don't want the results to be sorted.
+    - `lodash` was upgraded to `~3.10.1`. Many changes have occurred. Some of which
```
Comment (jdalton) [inline CHANGELOG]: I [opened a PR on grunt-legacy-util](https://github.com/gruntjs/grunt-legacy-util/pull/13) to bump lodash to v4.

Comment (shama) [inline CHANGELOG]: Thanks @jdalton!

@vladikoff I've bumped and integrated. Will finish up additional testing this evening. The release notes draft has been updated for it.

**Comment by shama** [comment]
Merging as appveyor mistakenly marked the build as failing even though all the tests passed: https://ci.appveyor.com/project/gruntjs/grunt/build/100/job/x0d1geuqhsalo4up

**Pull request merged**